manifests/init.pp

```diff
@@ -146,7 +146,7 @@
   Boolean $syslog                                                                            = false,
   Variant[Stdlib::Absolutepath, Boolean[false], Undef] $pidfile                              = $memcached::params::pidfile,
   Boolean $manage_firewall                                                                   = false,
-  String[1] $max_memory                                                                      = '95%',
+  Variant[Integer[0], Pattern[/^1?\d?\d%$/]] $max_memory                                     = '95%',
```

Comment:
Alternatively we can use String[1] instead of Pattern but I want to here any opinion from the others about the strict pattern vs the vague type.